src/components/CreatePledgeForm.jsx

```diff
@@ -1,77 +1,101 @@
 import { useState } from "react";
 import { useNavigate } from "react-router";
-
+import { useParams } from "react-router-dom";
 import postPledge from "../api/post-createpledge";
 
 
 function CreatePledgeForm() {
     const navigate = useNavigate();
+    const userId = localStorage.getItem("userId");
```

Comment:
I have also changed the log in form to set the userId based upon the changes Jordan helped everyone with the other night, here we access that so we can pass it into your post request.